### PR TITLE
cane gun part 2, this time it works

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
add:adds cane gun for based mime and chad clown as role specific traitor item

/:cl:

## About The Pull Request
fortnite

## Why It's Good For The Game
poopshit

### CODE REVIEW

**Comment by yoyobatty** [comment]
Put adds the cane gun instead of cane gun as it will literally show up as cane gun in the changelog.

**Comment by karmaisblackandbluepilled** [comment]
done

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 1]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+	righthand_file = 'hippiestation/icons/mob/inhands/righthand.dmi'
+	force = 15
+	throwforce = 7
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("pimped", "smacked", "disciplined", "busted", "capped", "decked")
+	resistance_flags = FIRE_PROOF
+	rack_sound_volume = 0
+	semi_auto = TRUE
+
+var/mob/current_owner
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Uh

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 2]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+
+/obj/item/gun/ballistic/shotgun/canegun/equipped(mob/user, slot)
+	slot = SLOT_HANDS
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Uh

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 3]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+	var/mob/holder = get(loc, /mob)
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Uh

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 4]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+	var/mob/living/carbon/human/H = user
+	if(!current_owner && holder)
+		current_owner = holder
+		if(istype(H))
+			if(H.get_num_legs(1) < 2)
+				H.add_movespeed_modifier(type, update=TRUE, priority=100, multiplicative_slowdown=-3, blacklisted_movetypes=(FLYING|FLOATING))
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Uh

**Comment by HideAndSeekLOGIC** [comment]
Did you test this?

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 5]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+	if(current_owner && current_owner != holder)
+		H.remove_movespeed_modifier(type)
+		current_owner = null
+		. = ..()
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Uh

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 6]
```diff
@@ -51,3 +51,51 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+
+/obj/item/gun/ballistic/shotgun/canegun/sawoff(mob/user)
+	to_chat(user, "<span class='warning'>Kinda defeats the purpose of a cane, doesn't it?</span>")
+	return
+
+/obj/item/pimpstick/suicide_act(mob/user)
+		user.visible_message("<span class='suicide'>[user] is hitting [user.p_them()]self with [src]! It looks like [user.p_theyre()] trying to discipline [user.p_them()]self for being a mark-ass trick.</span>")
+		return (BRUTELOSS)
```
Comment (yoyobatty) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: The indentation here

**Comment by HideAndSeekLOGIC** [comment]
Yeah, I don't think I'm approving this until this is tested to work properly. There are so many errors here I don't even know what to say.

**Comment by karmaisblackandbluepilled** [comment]
This works perfectly, I tested it on a private server. I literally do not know how to make it differently.
Saying uh doesn't tell me what is actually wrong.

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 2 above)
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: What is this? Don't you want
```
if(slot == SLOT_HANDS)
```

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 1 above)
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: This isn't indented, making it a crappy global var.

Make it a part of the item.

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 3 above)
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: What are you doing here?
You have `current_owner` already, and there's the `user` var for the new holder.

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm (hunk 4 above)
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: Seems a bit fast

Comment (karmaisblackandbluepilled) [reply hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: that's the speed you gotta add to make the 1-legged nigga walk like a normal person

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 7]
```diff
@@ -51,3 +51,56 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+				if(H.get_num_legs(1) < 2)
+					H.add_movespeed_modifier(type, update=TRUE, priority=100, multiplicative_slowdown=-3, blacklisted_movetypes=(FLYING|FLOATING))
+		if(current_owner && current_owner != user)
+			H.remove_movespeed_modifier(type)
+			current_owner = null
+	. = ..()
+
+/obj/item/gun/ballistic/shotgun/canegun/dropped(mob/user, slot)
+	var/mob/living/carbon/human/H = user
+	H.remove_movespeed_modifier(type)
```
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: should probably do an `ishuman` check before casting and removing

File: hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm [hunk 8]
```diff
@@ -51,3 +51,56 @@ obj/item/gun/ballistic/shotgun/triplebarrel/sawoff(mob/user)
...
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb = list("pimped", "smacked", "disciplined", "busted", "capped", "decked")
+	resistance_flags = FIRE_PROOF
+	rack_sound_volume = 0
+	semi_auto = TRUE
+
+
+
+/obj/item/gun/ballistic/shotgun/canegun/equipped(mob/user, slot)
+	var/mob/current_owner
```
Comment (ghost) [inline hippiestation/code/modules/projectiles/guns/ballistic/shotgun.dm]: should be below `semi_auto

**Comment by steamp0rt** [comment]
it's an old hippie port i think, plus it has +2 from designers, so i'll merge this as soon as it passes travis and co.

**Pull request merged**